src/lib/azure/azure.ts

```diff
@@ -0,0 +1,179 @@
+import * as Core from '../../core';
+import * as Errors from '../../error';
+import OpenAI, { ClientOptions } from 'openai/index';
+import { RequestInit } from 'node-fetch';
+
+/** API Client for interfacing with the Azure OpenAI API. */
+export interface AzureClientOptions extends ClientOptions {
+  /**
+   * Defaults to process.env['OPENAI_API_VERSION'].
+   */
+  apiVersion?: string | undefined;
+  /**
+   * Your Azure endpoint, including the resource, e.g. `https://example-resource.azure.openai.com/`
+   */
+  endpoint?: string | undefined;
+  /**
+   * A model deployment, if given, sets the base client URL to include `/deployments/{deployment}`.
+   * Note: this means you won't be able to use non-deployment endpoints. Not supported with Assistants APIs.
+   */
+  deployment?: string | undefined;
+  /**
+   * Defaults to process.env['AZURE_OPENAI_API_KEY'].
+   */
+  apiKey?: string | undefined;
+  /**
+   * A function that returns a Microsoft Entra (formerly known as Azure Active Directory) access token, will be invoked on every request.
+   */
+  azureEntraTokenProvider?: (() => string) | undefined;
+}
+
+/** API Client for interfacing with the Azure OpenAI API. */
+export class AzureOpenAI extends OpenAI {
+  private _azureEntraTokenProvider: (() => string) | undefined;
+  /**
+   * API Client for interfacing with the Azure OpenAI API.
+   *
+   * @param {string | undefined} [opts.endpoint=process.env['OPENAI_API_VERSION'] ?? undefined]
+   * @param {string | undefined} [opts.apiVersion=process.env['AZURE_OPENAI_ENDPOINT'] ?? undefined] - Your Azure endpoint, including the resource, e.g. `https://example-resource.azure.openai.com/`
+   * @param {string | undefined} [opts.apiKey=process.env['AZURE_OPENAI_API_KEY'] ?? undefined]
+   * @param {string | undefined} opts.deployment - A model deployment, if given, sets the base client URL to include `/deployments/{deployment}`.   * @param {string | null | undefined} [opts.organization=process.env['OPENAI_ORG_ID'] ?? null]
+   * @param {string} [opts.baseURL=process.env['OPENAI_BASE_URL']] - Sets the base URL for the API.
+   * @param {number} [opts.timeout=10 minutes] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
+   * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
+   * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
+   * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
+   * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
+   * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
+   * @param {boolean} [opts.dangerouslyAllowBrowser=false] - By default, client-side use of this library is not allowed, as it risks exposing your secret API credentials to attackers.
+   */
```

Comment:
I don't believe we generally use jsdoc in this library, preferring docstrings in TypeScript types.

Reply:
I was following the style used in the OpenAI client: https://github.com/openai/openai-node/blob/bb4bce30ff1bfb06d9d6c7a185d952121bb716e2/src/index.ts#L91-L104 but I can delete it if it doesn't break any tooling.

Reply:
Ah, my mistake. Fine to leave then.